front/src/reducers/osrdsimulation/types.ts

```diff
@@ -93,13 +93,45 @@ export interface Regime {
   mechanical_energy_consumed: number;
 }
 
-export interface ElectrificationConditions {
+// export interface ElectrificationConditions {
+//   start: number;
+//   stop: number;
+//   used_mode: string;
+//   used_profile: string;
+//   used_restriction?: string;
+//   seen_restriction?: string;
+// }
+
+export interface PowerRestrictionRange {
+  start: number;
+  stop: number;
+  code: string;
+  handled: boolean;
+}
+
+export interface ElectrificationRange {
   start: number;
   stop: number;
-  used_mode: string;
-  used_profile: string;
-  used_restriction?: string;
-  seen_restriction?: string;
+  electrificationUsage: Electrified | NonElectrified | Neutral;
+}
+
+export interface Electrified {
+  mode: string;
+  mode_handled: boolean;
+  profile?: string;
+  profile_handled: boolean;
+  object_type: 'Electrified';
+}
+
+export interface Neutral {
+  is_lower_pantograph: boolean;
+  object_type: 'Neutral';
+}
+
+export interface NonElectrified {
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  type: Object;
```

Comment:
for the TS types we never use `Object`, `Number`, `String`, `Symbol`, `Array` but the lowercase counterpart
```suggestion
  type: object;
```

Reply:
Done